uclient-fetch.c

```diff
@@ -339,6 +349,13 @@ static int init_request(struct uclient *cl)
 
 	uclient_http_reset_headers(cl);
 	uclient_http_set_header(cl, "User-Agent", user_agent);
+
+	list_for_each_entry_safe(h, th, &headers, list) {
+		uclient_http_set_header(cl, h->name, h->value);
```

Comment:
How does this handle POST data, if I want `Content-Type: application/json`?  Looks like it would be overridden by the code down at line 363.

Reply:
So, maybe if there are any user-supplied `--headers` present, then disable the `post_xxx` sections from adding further ones?  Seems like that would be pretty much backward compatible.

Reply:
The GNU wget allows overriding of a Content-Type when the `--post-data` is used.
Generally speaking this must be prohibited: the `--post-data` is to imitate a form post. If you wish to send a body with a custom CT e.g. `apllication/json` then you must use the `--methodn=POST --body-data` introduced in the https://github.com/openwrt/uclient/pull/2

To implement the CT overriding we need more code: introduce a `ct_header` var, check each supplied header name for the name and set it there, then check if the `ct_header` is NULL and if wasn't set then use the `application/x-www-form-urlencoded`.